gdb/mips-tdep.c

```diff
@@ -296,7 +296,7 @@ mips_simple_cheri_register_p (struct gdbarch *gdbarch, int regnum)
   int cap0 = mips_regnum (gdbarch)->cap0;
 
   return (cap0 != -1
-	  && rawnum >= cap0 && rawnum <= mips_regnum (gdbarch)->cap_pcc);
+	  && rawnum >= cap0 && rawnum < mips_regnum (gdbarch)->cap_cause);
```

Comment:
I believe this only gets used for the builtin XML features btw, but this is either way.